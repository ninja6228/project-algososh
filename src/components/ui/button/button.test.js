import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './button'

describe('Тестирование компонента Button', () => {
  it('с текстом', () => {
    const button = renderer.create(<Button text='Текст проверки' />)
    expect(button).toMatchSnapshot();
  });

  it('без текста', () => {
    const button = renderer.create(<Button />)
    expect(button).toMatchSnapshot();
  });

  it('в заблокированном состояние', () => {
    const button = renderer.create(<Button disabled={true} />)
    expect(button).toMatchSnapshot();
  });

  it('с индикацией загрузки', () => {
    const button = renderer.create(<Button isLoader={true} />)
    expect(button).toMatchSnapshot();
  });

  it('на вызов колбека при клике', () => {
    window.alert = jest.fn()
    const click = alert('Нажали кнопку');
    render(<Button text={'Кнопка для колбека'} onClick={click} />)
    const button = screen.getByText('Кнопка для колбека')
    fireEvent.click(button)
    expect(window.alert).toHaveBeenLastCalledWith('Нажали кнопку')
  })
})
