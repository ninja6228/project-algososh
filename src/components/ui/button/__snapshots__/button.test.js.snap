// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Button без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button в заблокированном состояние 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование компонента Button с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст проверки
  </p>
</button>
`;
